net/strongswan/files/ipsec.init

```diff
@@ -62,13 +62,15 @@ add_crypto_proposal() {
 	local encryption_algorithm
 	local hash_algorithm
 	local dh_group
+	local prf_algorithm
 
 	config_get encryption_algorithm  "$1" encryption_algorithm
 	config_get hash_algorithm        "$1" hash_algorithm
 	config_get dh_group              "$1" dh_group
+	config_get prf_algorithm         "$1" prf_algorithm
 
 	[ -n "$encryption_algorithm" ] && \
-		crypto="${crypto:+${crypto},}${encryption_algorithm}${hash_algorithm:+-${hash_algorithm}}${dh_group:+-${dh_group}}"
+		crypto="${crypto:+${crypto},}${encryption_algorithm}${hash_algorithm:+-${hash_algorithm}}${dh_group:+-${dh_group}}${prf_algorithm:+-${prf_algorithm}}"
```

Comment:
prf needs to go after hash-algorithm

Also, make sure the hash algorithm is not defined if an AEAD Algorithm is used, but then you have to define the prf manually. By default it's the same algorithm as the hmac (but used as prf instead). But if no hmac is defined, then the prf has to be specified manually.